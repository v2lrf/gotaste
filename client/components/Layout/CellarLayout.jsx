import React, { Fragment } from 'react'
import PropTypes from 'prop-types'
import Head from 'next/head'
import Link from 'next/link'

import Container from '../Container'
import NavItem from '../NavBar/NavItem'

function CellarLayout({ title, children }) {
  return (
    <Fragment>
      <Head>
        <title>{title}</title>
      </Head>
      <div className="bg-white">
        <Container>
          <div className="h-auto sm:h-16 flex items-center justify-between">
            <div className="my-4 sm:my-0">
              <Link href="/">
                <a className="flex text-red-light text-xl font-semibold no-underline">
                  <img
                    src="/static/gotaste-120.png"
                    alt="GoTaste"
                    className="h-6 pr-2"
                  />
                  GoTaste Cellar
                </a>
              </Link>
            </div>
            <ul className="list-reset flex flex-row">
              <NavItem asActive to="/cellar">
                Overblik
              </NavItem>
              <NavItem asActive to="#">
                Analytics
              </NavItem>
              <NavItem asActive to="#">
                Begivenheder
              </NavItem>
              <NavItem asActive to="#">
                Anmeldelser
              </NavItem>
              <NavItem asActive to="/cellar/profile">
                Profil
              </NavItem>
              <div className="mx-2 my-1 border border-red-lightest" />
              <NavItem to="/sign-out">Log ud</NavItem>
            </ul>
          </div>
        </Container>
      </div>
      {children}
    </Fragment>
  )
}

CellarLayout.propTypes = {
  title: PropTypes.string,
  children: PropTypes.node.isRequired
}

CellarLayout.defaultProps = {
  title: 'GoTaste Cellar'
}

export default CellarLayout
