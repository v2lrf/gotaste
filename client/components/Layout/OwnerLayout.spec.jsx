import React from 'react'
import CellarLayout from './CellarLayout'

describe('CellarLayout', () => {
  let component
  beforeEach(() => {
    component = shallow(<CellarLayout>Some content!</CellarLayout>)
  })

  describe('Snapshots', () => {
    it('should match the snapshot', () => {
      expect(component).toMatchSnapshot()
    })
  })
})
