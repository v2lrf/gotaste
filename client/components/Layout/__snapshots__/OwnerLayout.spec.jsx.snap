// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CellarLayout Snapshots should match the snapshot 1`] = `
<Fragment>
  <Head>
    <title>
      GoTaste Cellar
    </title>
  </Head>
  <div
    className="bg-white"
  >
    <Container
      narrow={false}
    >
      <div
        className="h-auto sm:h-16 flex items-center justify-between"
      >
        <div
          className="my-4 sm:my-0"
        >
          <Link
            href="/"
          >
            <a
              className="flex text-red-light text-xl font-semibold no-underline"
            >
              <img
                alt="GoTaste"
                className="h-6 pr-2"
                src="/static/gotaste-120.png"
              />
              GoTaste Cellar
            </a>
          </Link>
        </div>
        <ul
          className="list-reset flex flex-row"
        >
          <withRouter(NavItem)
            asActive={true}
            to="/cellar"
          >
            Overblik
          </withRouter(NavItem)>
          <withRouter(NavItem)
            asActive={true}
            to="#"
          >
            Analytics
          </withRouter(NavItem)>
          <withRouter(NavItem)
            asActive={true}
            to="/cellar/events"
          >
            Begivenheder
          </withRouter(NavItem)>
          <withRouter(NavItem)
            asActive={true}
            to="#"
          >
            Anmeldelser
          </withRouter(NavItem)>
          <withRouter(NavItem)
            asActive={true}
            to="/cellar/profile"
          >
            Profil
          </withRouter(NavItem)>
          <div
            className="mx-2 my-1 border border-red-lightest"
          />
          <withRouter(NavItem)
            to="/sign-out"
          >
            Log ud
          </withRouter(NavItem)>
        </ul>
      </div>
    </Container>
  </div>
  Some content!
</Fragment>
`;
